parsec/core/gui/files_widget.py

```diff
@@ -70,6 +70,101 @@ async def _do_delete(workspace_fs, files, silent=False):
+        base_name = pathlib.Path(src.name)
+        # In order to be able to rename the file if a file of the same name already exists
+        # we need the name without extensions.
+        # .stem only removes the first extension, so we loop over it.
+        while str(base_name) != base_name.stem:
+            base_name = pathlib.Path(base_name.stem)
+        count = 2
+        base_name = str(base_name)
+        while True:
+            try:
+                dst = current_directory / file_name
+                if src_type == FileType.Folder:
+                    await workspace_fs.copytree(src, dst, source_workspace)
+                else:
+                    await workspace_fs.copyfile(src, dst, source_workspace)
+                break
+            except FileExistsError:
+                # File already exists, we append a counter at the end of its name
+                file_name = "{} ({}){}".format(
+                    base_name, count, "".join(pathlib.Path(src.name).suffixes)
+                )
+                count += 1
+            except FSInvalidArgumentError as exc:
+                # Move a file onto itself
+                # Not a big deal for files, we just do nothing and pretend we
+                # actually did something
+                # For folders we have to warn the user
+                if src_type == FileType.Folder:
+                    error_count += 1
+                    last_exc = exc
+                break
+            except Exception as exc:
+                # No idea what happened, we'll just warn the user that we encountered an
+                # unexcepted error and log it
+                error_count += 1
+                last_exc = exc
+                logger.exception("Unhandled error while cut/copy file", exc_info=exc)
+                break
+    if error_count:
+        raise JobResultError("error", last_exc=last_exc, error_count=error_count)
+
+
+async def _do_move_files(workspace_fs, current_directory, files, source_workspace):
+    error_count = 0
+    last_exc = None
+    for f in files:
+        src = f[0]
+        src_type = f[1]
+        file_name = src.name
+        base_name = pathlib.Path(src.name)
+        # In order to be able to rename the file if a file of the same name already exists
+        # we need the name without extensions.
+        # .stem only removes the first extension, so we loop over it.
+        while str(base_name) != base_name.stem:
+            base_name = pathlib.Path(base_name.stem)
+        count = 2
+        base_name = str(base_name)
+        while True:
+            try:
+                dst = current_directory / file_name
+                await workspace_fs.move(src, dst, source_workspace)
+                break
+            except FileExistsError:
+                # File already exists, we append a counter at the end of its name
+                file_name = "{} ({}){}".format(
+                    base_name, count, "".join(pathlib.Path(src.name).suffixes)
```

Comment:
base_name is only used here, hence would be better to define it there too

Reply:
Not sure what you mean. It seems to me that it's better to get the file name without extensions before the loop instead of doing the same split over and over again inside the loop.

Reply:
Given the while loop should do very few iteration (most of the time no iterations at all !) I don' think there is any performance issue.
On the other hand, this splitting logic put before the loop decrease the readability for the main case.

However now that the splitting logic is a single trivial one-liner, I think it's less useful than when the comment was added. So do as you feel ;-)